lib/dialects/mysql/index.js

```diff
@@ -191,6 +198,60 @@ class Client_MySQL extends Client {
       options: {},
     });
   }
+
+  async checkVersion(connection) {
+    if (this._versionCheckPromise) {
```

Comment:
Minor: since we're already mutating connection objects, it wouldn't really be any worse for us to store the version on the connection instead of the client.

We're not necessarily guaranteed that every connection ever returned by the pool is the same database, so doing it this way means that if, e.g., an operator fails over a database to a backup that's running an older version, knex may start doing the wrong thing by assuming a newer version than it is now connected to.

(minor because this appears to be how the other dialects do it, but I think it's still worth a mention)

Reply:
I admit it, I am torn. So...

You are right - in theory. Yes, in theory the db could potentially change under the library's butt; however, I also think it would be *really* bad practice, to have a different (!) / lesser (?!?) version as an alternative. I've been around a while, and I've never -- ever -- seen that. If I *did* see it, I would probably question what the DBA was drinking (smoking?) when he set it up, and ask for some.

I don't know. I guess I am resisting this, since it would mean 1) Store the version in the connection, change the tests 2) Open an issue so that we fix other dialects 3) Understand that we are making one extra query per connection, every single connection, to ask something that really... really shouldn't change.

I might be missing something though.

Reply:
I just realised I am muddling things up. Moving things to the connection is linked to the "do we exit on versioning regex problems", but it solves a different issue: "DO we cover the case where a connection within the same app happens to be on a lesser/different DB version", in which case I say "No, I would opt to sack the DBA in that case"...

Reply:
Yeah, this thread is resolved. I just wanted your thoughts since we had the opportunity to do this one in a potentially better way.

Reply:
Wait wait, if you think it's worthwhile doing, it probably is. At the very least, I will open an issue where we can discuss this properly and decide there -- agreed?